x-pack/plugins/monitoring/dev_docs/reference/data_collection_modes.md

```diff
@@ -0,0 +1,199 @@
+Stack monitoring has three metrics collection modes:
+
+- Previous standard, still supported: [Internal collection](#internal-collection)
+- Current standard: [Standalone Metricbeat collection](#standalone-metricbeat-collection)
+- Planned future standard: [Package-driven collection](#package-driven-collection)
+
+And two log collection modes:
+
+- Current standard: [Filebeat log collection](#logs-collection-filebeat)
+- Planned future standard: [Package-driven collection](#package-driven-collection)
+
+> **Note**: In the following sections "Beats" is used to refer to any stack component based on libbeat because the data collection modes are the same for any process in this group. This includes apm-server.
+ 
+## Metrics collection
+
+### Internal collection
+
+When using internal collection, each component in your Elastic stack will collect its own data via its own mechanism. In general this is an in-memory structure, but the implementation details vary per component.
+
+On a cycle (typically 10 seconds), each stack component then typically publishes the internally collected metrics to the [production deployment](../reference/terminology.md#production-deployment).
+
+Here's a diagram using kibana as an example. Other stack components have similar collection and publishing mechanisms.
+
+```mermaid
+graph TD
+subgraph Production Deployment
+  ProdElasticsearch[Elasticsearch]
+end
+
+subgraph Monitoring Deployment
+  MonElasticsearch[Elasticsearch]
+end
+
+subgraph Kibana
+  Collectors-.->|poll|Metrics[OS, Cgroups, etc]
+  Collectors-.->|update|OpsMetricsObservable
+  BulkUploader-.->|observe|OpsMetricsObservable
+  
+  click Collectors "https://github.com/elastic/kibana/tree/main/src/core/server/metrics/collectors"
+  click OpsMetricsObservable "https://github.com/elastic/kibana/blob/92a8636f0ff63ab072527574e96e6616327b2ea4/src/core/server/metrics/metrics_service.ts#L32"
+  click BulkUploader "https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.ts"
+end 
+
+BulkUploader-->|_monitoring_bulk|ProdElasticsearch
```

Comment:
Below you say "The production cluster receives these metrics on the `/_monitoring/bulk` endpoint in elasticsearch" and here in the diagram its called `_monitoring_bulk`? Can you clarify for me?

Reply:
Ah, thanks! Yeah, pretty sure it's `/_monitoring/bulk`. Will check and make things consistent.